src/js/components/Diagram/Diagram.js

```diff
@@ -60,59 +60,56 @@ const findTarget = target => {
   return target;
 };
 
-class Diagram extends Component {
-  static defaultProps = { connections: [] };
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    // track whether the connections array changes so we can trigger re-placing
-    if (nextProps.connections !== prevState.connections) {
-      return {
-        connections: nextProps.connections,
-        connectionPoints: undefined,
-      };
-    }
-    return null;
-  }
-
-  state = { height: 0, width: 0 };
-
-  svgRef = React.createRef();
-
-  componentDidMount() {
-    window.addEventListener('resize', this.onResize);
-    this.onResize();
-  }
-
-  componentDidUpdate() {
-    const { connectionPoints } = this.state;
-    if (!connectionPoints) {
-      this.placeConnections();
-    }
+const Diagram = ({ connections, theme, ...rest }) => {
```

Comment:
```suggestion
const Diagram = ({ connections = [], theme, ...rest }) => {
```
moving up from default props

Reply:
I just did it again anyway. It fails a test in [src/js/components/__tests__](https://github.com/grommet/grommet/blob/master/src/js/components/__tests__/components-test.js)

I have no idea why.. I mentioned this above also.

Reply:
yea, you are right, sorry for the hustle. I'll need to take a look at it myself, can you roll back your PR to have the array with the default props?

Reply:
Pushed a new commit, I don't know how to rollback a commit. I'm a very newbie!

Reply:
no worries, you are doing great my friend, at the end, it is all squashed anyway :)